### ISSUE TITLE
Not Connecting

### ISSUE THREAD

**Issue by ZacharyAiley** [bitfocus/companion-module-tplink-kasasmartdimmer#10]
Hello, I've purchased a new KS225 dimmer. It's fully updated to version "1.1.1" but it's not connecting. I am able to ping the device, but in the companion logs it's showing connection refused. I'm curious if these devices have different firmware now, since it's fully updated to 1.1.1.

**Comment by Remulys22** [comment]
I am also having this issue. trying to connect to the switches and getting a connection refused error. I am using the hs220 model though. Debug log attached. [module-log-11_2_2024-5_44_30 PM.csv](https://github.com/user-attachments/files/17608478/module-log-11_2_2024-5_44_30.PM.csv)

**Comment by jermakis** [comment]
I am also having issues with this connection.  My switches are KS225, firmware 1.1.1.  Log states connection refused:

error: TCP 192.168.1.4:9999 Error: connect ECONNREFUSED xxx.xxx.x.xxx:9999
    at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1555:16) {
  errno: -4078,
  code: 'ECONNREFUSED',
  syscall: 'connect',
  address: 'xxx.xxx.x.xxx',
  port: 9999
}

**Comment by josephdadams** [comment]
My understanding is that TP Link Kasa devices once you update the firmware to a certain point they have blocked the LAN access port 9999 so we can’t control them anymore.

**Comment by sonicSMASH** [comment]
Thinking of getting a Kasa dimmer, does this still work or not?

**Comment by jermakis** [comment]
Still not working, however I'm running home assistant and using BF Companion to toggle that way instead.

**Comment by sonicSMASH** [comment]
Thanks! Do i still get all of the functionality such as brightness output and feedback if using Home Assistant?

**Comment by jermakis** [comment]
Yes, there are a couple options built into the HA connection in BF Companion: adjust light brightness (percentage) and set light brightness (percentage).  I also use the feedback to show a color if the light is on/off.
